File: Data Structures & Algorithms/reorder-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public void reorderList(ListNode head) {

        // first i need to reverse the linked lIntStream
        // then after just loop through using one list vs the otherecord

        ListNode reverse = new ListNode(0);

        // so what i need to do is 
        // reverse the last half of the linked list or what i could do isealed
        // is reverse the entire list then just maintain a count

        ListNode slow = head;
        ListNode fast = head;
        while (fast.next != null && fast.next.next != null){

            fast = fast.next.next;
            slow = slow.next;

        }

        ListNode secondHalf = slow.next;
        slow.next = null;
        ListNode current = secondHalf; 
        ListNode prev = null;


        while (current != null){
            ListNode next = current.next;
            current.next = prev;
            prev = current;
            current = next;
        }

        ListNode finalList = head;

        while(prev != null){
            ListNode temp = head.next;
            ListNode prevTemp = prev.next;   
            head.next = prev;             
            prev.next = temp;             
            head = temp;                  
            prev = prevTemp;  
        }

    }
}

File: Data Structures & Algorithms/add-two-numbers/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:


        ## need to store as a string, reverse, add, reverse linked list

        l1_str = ""


        curr = l1

        while curr:
            l1_str += str(curr.val)
            curr = curr.next


        l1_reverse = l1_str[::-1]


        l2_str = ""


        curr = l2

        while curr:
            l2_str += str(curr.val)
            curr = curr.next

        l2_reverse = l2_str[::-1]

        sumOf = str(int(l1_reverse) + int(l2_reverse))
        sumOf = sumOf[::-1]


        i = 0

        dummy = ListNode(0)
        curr = dummy
            
        for char in str(sumOf):
            curr.next = ListNode(int(char))
            curr = curr.next
            
        return dummy.next
    




















